services/traefik/Makefile

```diff
@@ -0,0 +1,138 @@
+.DEFAULT_GOAL := help
+PREDEFINED_VARIABLES := $(.VARIABLES)
+
+# Operating system
+ifeq ($(filter Windows_NT,$(OS)),)
+IS_WSL  := $(if $(findstring Microsoft,$(shell uname -a)),WSL,)
+IS_OSX  := $(filter Darwin,$(shell uname -a))
+IS_LINUX:= $(if $(or $(IS_WSL),$(IS_OSX)),,$(filter Linux,$(shell uname -a)))
+endif
+IS_WIN  := $(strip $(if $(or $(IS_LINUX),$(IS_OSX),$(IS_WSL)),,$(OS)))
+$(info + Detected OS : $(IS_LINUX)$(IS_OSX)$(IS_WSL)$(IS_WIN))
+
+# Makefile's shell
+SHELL := $(if $(IS_WIN),powershell.exe,/bin/bash)
+
+DOCKER_COMPOSE=$(if $(IS_WIN),docker-compose.exe,docker-compose)
+DOCKER        =$(if $(IS_WIN),docker.exe,docker)
+
+# If you see pwd_unknown showing up, this is why. Re-calibrate your system.
+PWD ?= pwd_unknown
+# Internal VARIABLES ------------------------------------------------
+# STACK_NAME defaults to name of the current directory. Should not to be changed if you follow GitOps operating procedures.
+STACK_NAME = $(notdir $(PWD))
+SWARM_HOSTS = $(shell $(DOCKER) node ls --format={{.Hostname}} 2>/dev/null)
+TEMP_COMPOSE=.stack.${STACK_NAME}.yaml
+
+# External VARIABLES
+include .env
+
+# TARGETS --------------------------------------------------
+.PHONY: help
+help: ## This colourful help
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+.PHONY: up
+up: .init .env ${TEMP_COMPOSE} secrets/domain.key secrets/domain.crt ## Deploys the stack using provided certificates
+	$(DOCKER) stack deploy -c ${TEMP_COMPOSE} ${STACK_NAME}
+
+.PHONY: down
+down: ## Removes the stack from the swarm
+	$(DOCKER) stack rm ${STACK_NAME}
+
+.PHONY: leave
+leave: ## Leaves swarm stopping all services in it
+	-$(DOCKER) swarm leave -f
+
+.PHONY: clean
+clean: ## Cleans unversioned files
+	@git clean -dxf
+
+.PHONY: info
+info: ## expands all variables and relevant info on stack
```

Comment:
Minor: Looks totally great! I was just wondering whether we should do a fast review to all the makefiles to homogenize them (e.g. name of variables, targets, etc) within osparc-ops repo.

Same applies to configuration files e.g. `.env` vs `.env.config` etc etc